syntax = "proto3";

package vryndara;

message AgentInfo {
    string id = 1;
    repeated string capabilities = 2;
    string status = 3;
}

message Signal {
    string id = 1;
    string source_agent_id = 2;
    string target_agent_id = 3;
    string type = 4;
    string payload = 5;
    int64 timestamp = 6;
}

message Ack {
    bool success = 1;
    string error = 2;
}

service Kernel {
    rpc Register (AgentInfo) returns (Ack);
    rpc Publish (Signal) returns (Ack);
    rpc Subscribe (AgentInfo) returns (stream Signal);
}